Sources/StreamChat/Controllers/EventsController/ChannelEventsController.swift

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public extension _ChatClient {
+    /// Creates a new `ChannelEventsController` that can be used to listen to system events
+    /// related to the channel with `cid` and to send custom events.
+    ///
+    /// - Parameter cid: A channel identifier.
+    /// - Returns: A new instance of `ChannelEventsController`.
+    func channelEventsController(for cid: ChannelId) -> ChannelEventsController {
+        .init(
+            cid: cid,
+            eventSender: .init(database: databaseContainer, apiClient: apiClient),
+            notificationCenter: eventNotificationCenter
+        )
+    }
+}
+
+public extension _ChatChannelController {
+    /// Creates a new `ChannelEventsController` that can be used to listen to system events
+    /// and for sending custom events into a channel the current controller manages.
+    ///
+    /// - Throws: `ChannelNotCreatedYet` if channel hasn't yet been created.
+    /// - Returns: A new instance of `ChannelEventsController`.
+    func eventsController() throws -> ChannelEventsController {
+        guard let cid = cid else { throw ClientError.ChannelNotCreatedYet() }
+        
+        return .init(
+            cid: cid,
+            eventSender: .init(database: client.databaseContainer, apiClient: client.apiClient),
+            notificationCenter: client.eventNotificationCenter
+        )
+    }
```

Comment:
we have a discussion with @tbarbugli that it'd be nice to make this func not throwing 👍 Looking how we can approach this

Reply:
resolved by switching to cid provider closure